From: "Matteo F. Vescovi" <mfv@debian.org>
Date: Sun, 7 Sep 2025 15:20:01 +0200
Subject: locales_directory_install

Change where locales are installed.

Contributions by:
 Antonio Ospite <ao2@ao2.it>
 Gleb Fotengauer-Malinovskiy <glebfm@altlinux.org>
 Giovanni Panozzo <giovanni@panozzo.it>
---
 source/blender/blenkernel/BKE_appdir.hh           | 1 +
 source/blender/blenkernel/intern/appdir.cc        | 3 +++
 source/blender/blentranslation/intern/blt_lang.cc | 4 ++--
 source/creator/CMakeLists.txt                     | 2 +-
 4 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/source/blender/blenkernel/BKE_appdir.hh b/source/blender/blenkernel/BKE_appdir.hh
index d82bb33..d2b4b06 100644
--- a/source/blender/blenkernel/BKE_appdir.hh
+++ b/source/blender/blenkernel/BKE_appdir.hh
@@ -168,6 +168,7 @@ enum {
   BLENDER_SYSTEM_SCRIPTS = 53,
   BLENDER_SYSTEM_EXTENSIONS = 54,
   BLENDER_SYSTEM_PYTHON = 55,
+  BLENDER_SYSTEM_LOCALE = 56,
 };
 
 /** For #BKE_appdir_folder_id_version only. */
diff --git a/source/blender/blenkernel/intern/appdir.cc b/source/blender/blenkernel/intern/appdir.cc
index c6cb161..45b4c40 100644
--- a/source/blender/blenkernel/intern/appdir.cc
+++ b/source/blender/blenkernel/intern/appdir.cc
@@ -706,6 +706,9 @@ bool BKE_appdir_folder_id_ex(const int folder_id,
       }
       return false;
 
+      case BLENDER_SYSTEM_LOCALE:
+        BLI_strncpy(path, "/usr/share/locale", FILE_MAX); break;
+
     default:
       BLI_assert_unreachable();
       break;
diff --git a/source/blender/blentranslation/intern/blt_lang.cc b/source/blender/blentranslation/intern/blt_lang.cc
index 32a379c..6ff2435 100644
--- a/source/blender/blentranslation/intern/blt_lang.cc
+++ b/source/blender/blentranslation/intern/blt_lang.cc
@@ -67,7 +67,7 @@ static void free_locales()
 
 static void fill_locales()
 {
-  std::optional<std::string> languages_path = BKE_appdir_folder_id(BLENDER_DATAFILES, "locale");
+  std::optional<std::string> languages_path = BKE_appdir_folder_id(BLENDER_SYSTEM_LOCALE, NULL);
   if (!languages_path.has_value()) {
     CLOG_WARN(&LOG, "'locale' data path for translations not found");
     return;
@@ -254,7 +254,7 @@ void BLT_lang_set(const char *str)
   std::string locale_name = str ? str : LOCALE(ulang);
 
   /* #blender::locale assumes UTF8, no need to put it in the name. */
-  const std::optional<std::string> messagepath = BKE_appdir_folder_id(BLENDER_DATAFILES, "locale");
+  const std::optional<std::string> messagepath = BKE_appdir_folder_id(BLENDER_SYSTEM_LOCALE, NULL);
   blender::locale::init(locale_name, {TEXT_DOMAIN_NAME}, {messagepath.value_or("")});
 
 #else
diff --git a/source/creator/CMakeLists.txt b/source/creator/CMakeLists.txt
index f0c81fc..c10809d 100644
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -502,7 +502,7 @@ install(
 # localization
 if(WITH_INTERNATIONAL)
   set(_locale_dir "${CMAKE_SOURCE_DIR}/locale")
-  set(_locale_target_dir ${TARGETDIR_VER}/datafiles/locale)
+  set(_locale_target_dir ${CMAKE_INSTALL_PREFIX}/share/locale)
 
   file(GLOB _po_files "${_locale_dir}/po/*.po")
   foreach(_po_file ${_po_files})
